controllers/baseboardmanagement_controller.go

```diff
@@ -18,42 +18,193 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	bmcv1alpha1 "github.com/tinkerbell/rufio/api/v1alpha1"
+	bmcclient "github.com/tinkerbell/rufio/pkg/bmc/client"
 )
 
 // BaseboardManagementReconciler reconciles a BaseboardManagement object
 type BaseboardManagementReconciler struct {
 	client.Client
```

Comment:
I don't think the `BaseboardManagementReconciler` has an 'is a' relationship with a client so this shouldn't be embedded. Embedding it exposes all the behavior of the client from the reconciler so it can be used in-place of a client which doesn't seem right.

Reply:
If i understood correctly you are referring to the `BMCClient`? client.Client is actually `controller-runtime client`.

If BMCClient, then I do agree, but I was not sure of a better way to inject the dependency into the controller? Since the BaseboardManagementReconciler needs to perform BMC operations I thought we could make the client a part of the struct.

Reply:
Nope. I'm referring to the fact there's a relationship between `BaseboardManagementReconciler` and `client.Client` and that relationship is seemingly wrong :). 

```go
// 'is a' type relationship.
type BaseboardManagementReconciler struct {
	client.Client 
}

// 'has a' type relationship.
type BaseboardManagementReconciler struct {
	client client.Client
}
```

Reply:
Ahh my bad! So I actually didnt think to change that because it came with the auto-generated code.
Thanks for the detailed explanation. It makes sense now😄